### DISCUSSION TITLE
some module ideas

### DISCUSSION THREAD

**Discussion by Koto-Tori** [etianl/Trouser-Streak#10] [Ideas]
hii im new to github and stuff so i hope this is in the right place. id like to suggest some modules that ive seen in other addons/clients,

- oresim, meteor rejects
- basefinder, wurst (or basically just something that could see what should be where, like oresim with the seed, and then cross reference that to try to find bases, esp in conjunction with new chunks)

idk how hard this stuff is to implement though lol ;w;

**Comment by etianl** [comment]
I appreciate the input and ideas, I don't think I will do the oresim thing. Use Rejects for that if you claim it's already in there, Meteor Rejects has many other fun modules to offer too such as instamine. The basefinder idea has got me thinking of something that I might work on when I have more spare time to code

**Reply by Koto-Tori** [reply]
oo okay, i hope it comes out good :)) thanks for responding

**Reply by etianl** [reply]
I have completed the first release of the BaseFinder module. If it needs updating I will fine tune it more. I hope you enjoy :)

**Reply by Koto-Tori** [reply]
wow cool!! ill try it out later, thanks so much :))

**Reply by etianl** [reply]
It will certainly need some fine tuning but it does find most bases unless they are really tiny like a noob just barely starting to play

**Comment by Jumima68** [comment]
Would it be possible to add a option to toggle showing natural spawners, so only spawners that are placed by players are shown?

**Reply by etianl** [reply]
I don't think it's possible to detect spawners placed just by players. The only thing that I could think of would be to detect spawners only if they are of an unnatural type (pig spawner, wither skele spawner, etc.), with the way it's written now that'd be hard to implement. There is also the other thing where if there is more than 1 or a few spawners in a chunk then it's likely not natural. (except for the very rare times it is natural). Try putting mob spawners on list three, and see if that makes any false positives for you. If no false positive, then that'd be a good place to  add mob spawners, if the list three value is left at 4.  (It'll look for all the things from that list and when the total number of things found in that chunk =4 then it'll throw a base location)

**Reply by Jumima68** [reply]
The solution could be that it looks if mossy cobble stone is in a five block radius, for dungeons and for cave spiders u it could look after cobwebs. These blocks are 100% near the spawners. I dont know if thats possible but if it is it could use the Block Esp to detect the blocks.

**Reply by etianl** [reply]
This idea has been written in, as an option called "Unnatural Spawner Finder", also block scanning performance has been improved. Changes are coming in Trouser version 0.6.0 (release soon, I am working on some other stuff too)

**Reply by etianl** [reply]
0.6.0 has been posted, with this feature implemented :)

**Comment by Vperah** [comment]
is it also possible to add different colors for different lists?

Like on list 1 I have shulkers and chests and a custom color (red)
On list 2 are spawners (blue)

so you can see what stuff you found

**Reply by etianl** [reply]
I was looking into this, and it looks like it would severely complicate the code so unfortunately I will not be doing this one unless I come up with an easy way to do it

**Reply by Vperah** [reply]
ok too bad, but still thanks for this wonderful mod ;D

**Reply by etianl** [reply]
Do not hesitate to post any other ideas you may have. I will seriously consider all ideas posted here so long as they will aid the griefing in some way
